### PR TITLE
Improve button demo - icons, predefinedTypes descriptions

### PR DESCRIPTION
**Pull request by czernous-devexpress**

### CODE REVIEW

**Review by RomanTsukanov — DISMISSED** [review]

File: JSDemos/Demos/Button/Icons/description.md [hunk 1]
```diff
@@ -1 +1,15 @@
-This demo illustrates how to&nbsp;display an&nbsp;icon within the **Button** UI component. You can use a&nbsp;built-in or&nbsp;external font, or&nbsp;use an&nbsp;image as&nbsp;an&nbsp;icon.
\ No newline at end of file
+To add an icon to **Button**, use the UI component's **icon** property. This demo covers the following use cases of this property:
```
Comment (Abadzhev) [inline JSDemos/Demos/Button/Icons/description.md]:
```suggestion
To add an icon to a **Button**, set the **icon** property. This demo shows several ways you can use this property:   
```

File: JSDemos/Demos/Button/Icons/description.md [hunk 2]
```diff
@@ -1 +1,15 @@
-This demo illustrates how to&nbsp;display an&nbsp;icon within the **Button** UI component. You can use a&nbsp;built-in or&nbsp;external font, or&nbsp;use an&nbsp;image as&nbsp;an&nbsp;icon.
\ No newline at end of file
+To add an icon to **Button**, use the UI component's **icon** property. This demo covers the following use cases of this property:   
+
+- **Built-in icons**    
+DevExtreme comes with an [icon library](/Documentation/Guide/Themes_and_Styles/Icons/#Built-In_Icon_Library). Set the **icon** property to the relevant icon name.
```
Comment (Abadzhev) [inline JSDemos/Demos/Button/Icons/description.md]:
```suggestion
DevExtreme ships with its own [icon library](/Documentation/Guide/Themes_and_Styles/Icons/#Built-In_Icon_Library). Pick any icon and assign its name to the **icon** property.
```

File: JSDemos/Demos/Button/Icons/description.md [hunk 3]
```diff
@@ -1 +1,15 @@
-This demo illustrates how to&nbsp;display an&nbsp;icon within the **Button** UI component. You can use a&nbsp;built-in or&nbsp;external font, or&nbsp;use an&nbsp;image as&nbsp;an&nbsp;icon.
\ No newline at end of file
+To add an icon to **Button**, use the UI component's **icon** property. This demo covers the following use cases of this property:   
+
+- **Built-in icons**    
+DevExtreme comes with an [icon library](/Documentation/Guide/Themes_and_Styles/Icons/#Built-In_Icon_Library). Set the **icon** property to the relevant icon name.    
+
+- **Image icons**
```
Comment (Abadzhev) [inline JSDemos/Demos/Button/Icons/description.md]: Can we say "image file"? "Image icons" sounds redundant. 

```suggestion
- **Image file**    
```

File: JSDemos/Demos/Button/Icons/description.md [hunk 4]
```diff
@@ -1 +1,15 @@
...
\ No newline at end of file
+To add an icon to **Button**, use the UI component's **icon** property. This demo covers the following use cases of this property:   
+
+- **Built-in icons**    
+DevExtreme comes with an [icon library](/Documentation/Guide/Themes_and_Styles/Icons/#Built-In_Icon_Library). Set the **icon** property to the relevant icon name.    
+
+- **Image icons**    
+Set the **icon** value to the image file path or URI.    
+
+- **External icon fonts**
```
Comment (Abadzhev) [inline JSDemos/Demos/Button/Icons/description.md]: I want to change "external" to "3rd-party". If possible, update the demo UI too. 

```suggestion
- **3rd-party icon fonts**
```

File: JSDemos/Demos/Button/Icons/description.md [hunk 5]
```diff
@@ -1 +1,15 @@
...
+To add an icon to **Button**, use the UI component's **icon** property. This demo covers the following use cases of this property:   
+
+- **Built-in icons**    
+DevExtreme comes with an [icon library](/Documentation/Guide/Themes_and_Styles/Icons/#Built-In_Icon_Library). Set the **icon** property to the relevant icon name.    
+
+- **Image icons**    
+Set the **icon** value to the image file path or URI.    
+
+- **External icon fonts**   
+It is possible to import an external font library. In this case, use **icon** names taken from the library. Refer to the library's documentation for proper icon names. This example uses <a href="https://fontawesome.com/icons?d=gallery" target="_blank">Font Awesome</a>.
```
Comment (Abadzhev) [inline JSDemos/Demos/Button/Icons/description.md]:
```suggestion
You can import a 3rd-party font library (this example uses <a href="https://fontawesome.com/icons?d=gallery" target="_blank">Font Awesome</a>). In such cases, set the **icon** property to a name that identifies the required glyph. Look up names in the imported library's documentation.
```

File: JSDemos/Demos/Button/Icons/description.md [hunk 6]
```diff
@@ -1 +1,15 @@
...
+- **Built-in icons**    
+DevExtreme comes with an [icon library](/Documentation/Guide/Themes_and_Styles/Icons/#Built-In_Icon_Library). Set the **icon** property to the relevant icon name.    
+
+- **Image icons**    
+Set the **icon** value to the image file path or URI.    
+
+- **External icon fonts**   
+It is possible to import an external font library. In this case, use **icon** names taken from the library. Refer to the library's documentation for proper icon names. This example uses <a href="https://fontawesome.com/icons?d=gallery" target="_blank">Font Awesome</a>.    
+
+- **Icon without a text**
```
Comment (Abadzhev) [inline JSDemos/Demos/Button/Icons/description.md]:
```suggestion
- **Buttons with icons and no caption text**   
```

File: JSDemos/Demos/Button/Icons/description.md [hunk 7]
```diff
@@ -1 +1,15 @@
...
+DevExtreme comes with an [icon library](/Documentation/Guide/Themes_and_Styles/Icons/#Built-In_Icon_Library). Set the **icon** property to the relevant icon name.    
+
+- **Image icons**    
+Set the **icon** value to the image file path or URI.    
+
+- **External icon fonts**   
+It is possible to import an external font library. In this case, use **icon** names taken from the library. Refer to the library's documentation for proper icon names. This example uses <a href="https://fontawesome.com/icons?d=gallery" target="_blank">Font Awesome</a>.    
+
+- **Icon without a text**   
+Define the **icon** but do not define the **text** property.
```
Comment (Abadzhev) [inline JSDemos/Demos/Button/Icons/description.md]: I feel the need to add information about tooltips here...

File: JSDemos/Demos/Button/PredefinedTypes/description.md [hunk 8]
```diff
@@ -1,3 +1,14 @@
-This demo illustrates the predefined types of&nbsp;the **Button** UI component. A&nbsp;[Toast][0] is&nbsp;displayed each time you click a&nbsp;**Button**.
+The **Button** supports several types that define its color. To specify the type, use the [type](/Documentation/ApiReference/UI_Widgets/dxButton/Configuration/#type) property. This property accepts the following values:
```
Comment (Abadzhev) [inline JSDemos/Demos/Button/PredefinedTypes/description.md]: The main reason for the change is that use the word "type" too many times. 

```suggestion
The **Button** widget supports several predefined color schemes. To apply a scheme, set the [type](/Documentation/ApiReference/UI_Widgets/dxButton/Configuration/#type) property to one of the following values:    
```

File: JSDemos/Demos/Button/PredefinedTypes/description.md [hunk 9]
```diff
@@ -1,3 +1,14 @@
...
 
-[0]: https://js.devexpress.com/Documentation/Guide/Widgets/Toast/Overview/
\ No newline at end of file
+- *"normal"*    
+- *"success"*
+- *"default"*    
+- *"danger"*    
+- *"back"*    
+
+To further customize the UI component, use the [stylingMode](/Documentation/ApiReference/UI_Widgets/dxButton/Configuration/#stylingMode) property. This property affects the fill and borders, and accepts the following values:
```
Comment (Abadzhev) [inline JSDemos/Demos/Button/PredefinedTypes/description.md]: I'm not sure about "UI Component". I'd probably just say "button" again.

File: JSDemos/Demos/Button/Icons/description.md [hunk 10]
```diff
@@ -1 +1,15 @@
-This demo illustrates how to&nbsp;display an&nbsp;icon within the **Button** UI component. You can use a&nbsp;built-in or&nbsp;external font, or&nbsp;use an&nbsp;image as&nbsp;an&nbsp;icon.
\ No newline at end of file
+To add an icon to a **Button**, set the **icon** property. This demo shows several ways you can use this property:
```
Comment (arminal) [inline JSDemos/Demos/Button/Icons/description.md]:
```suggestion
To add an icon to a **Button**, set the **icon** property. This demo shows how you can use this property:   
```

File: JSDemos/Demos/Button/Icons/description.md [hunk 11]
```diff
@@ -1 +1,15 @@
...
+DevExtreme ships with its own [icon library](/Documentation/Guide/Themes_and_Styles/Icons/#Built-In_Icon_Library). Pick any icon and assign its name to the **icon** property.
+
+- **Image file**    
+Set the **icon** value to the image file path or URI.    
+
+- **3rd-party icon fonts**    
+You can import a 3rd-party font library (this example uses <a href="https://fontawesome.com/icons?d=gallery" target="_blank">Font Awesome</a>). In such cases, set the **icon** property to a name that identifies the required glyph. Look up names in the imported library's documentation.
+
+- **Buttons with icons and no caption text**   
+Define the **icon** but do not define the **text** property. You can use the [hint](/Documentation/ApiReference/UI_Widgets/dxButton/Configuration/#hint) property to annotate the **button**.
```
Comment (arminal) [inline JSDemos/Demos/Button/Icons/description.md]:
```suggestion
Define the **icon**, but do not define the **text** property. You can use the [hint](/Documentation/ApiReference/UI_Widgets/dxButton/Configuration/#hint) property to annotate the **button**.   
```

**Review by RomanTsukanov — APPROVED** [review]

**Pull request merged**